src/main/java/evm/RepositoryMock.java

```diff
@@ -0,0 +1,173 @@
+                return false;
+            }
+        }
+
+        @Override
+        public void createAccount(byte[] address) {
+            if (!exists(address)) {
+                accounts.put(new ByteArrayWrapper(address), new Account());
+            }
+        }
+
+        @Override
+        public void delete(byte[] address) {
+            accounts.put(new ByteArrayWrapper(address), null);
+        }
+
+        @Override
+        public long increaseNonce(byte[] address) {
+            createAccount(address);
+            return getAccount(address).nonce += 1;
+        }
+
+        @Override
+        public long setNonce(byte[] address, long nonce) {
+            createAccount(address);
+            return (getAccount(address).nonce = nonce);
+        }
+
+        @Override
+        public long getNonce(byte[] address) {
+            Account account = getAccount(address);
+            return account == null ? 0 : account.nonce;
+        }
+
+        @Override
+        public void saveCode(byte[] address, byte[] code) {
+            createAccount(address);
+            getAccount(address).code = code;
+        }
+
+        @Override
+        public byte[] getCode(byte[] address) {
+            Account account = getAccount(address);
+            return account == null ? null : account.code;
+        }
+
+        @Override
+        public void putStorageRow(byte[] address, DataWord key, DataWord value) {
+            createAccount(address);
+            getAccount(address).storage.put(key, value);
+        }
+
+        @Override
+        public DataWord getStorageRow(byte[] address, DataWord key) {
+            Account account = getAccount(address);
+            return account == null ? null : account.storage.get(key);
+        }
+
+        @Override
+        public BigInteger getBalance(byte[] address) {
+            Account account = getAccount(address);
+            return account == null ? BigInteger.ZERO : account.balance;
+        }
+
+        @Override
+        public BigInteger addBalance(byte[] address, BigInteger value) {
+            createAccount(address);
+            Account account = getAccount(address);
+            return account.balance = account.balance.add(value);
+        }
+
+        @Override
+        public RepositoryMock startTracking() {
+            return new RepositoryMock(this);
+        }
+
+        @Override
+        public Repository clone() {
+            RepositoryMock copy = new RepositoryMock(parent);
+            for (Map.Entry<ByteArrayWrapper, Account> entry : accounts.entrySet()) {
+                copy.accounts.put(entry.getKey(), entry.getValue().clone());
+            }
+            return copy;
+        }
+
+        @Override
+        public void commit() {
+            if (parent != null) {
+                parent.accounts.putAll(accounts);
+            }
+        }
+
+        @Override
+        public void rollback() {
+            accounts.clear();
+        }
+
+        static class Account {
```

Comment:
We should have each class in a separate file, please move this one to a new file.

Reply:
Done the changes here 627b2cae92746b019c78bffb9d8ce8eaf3ec5157

Reply:
Also have added its java docs here 666620288578806a7dfec05bf1416c6f813218eb